src/error.rs

```diff
@@ -20,15 +32,9 @@ impl From<&str> for Error {
     }
 }
 
-impl From<serde_json::Error> for Error {
-    fn from(e: serde_json::Error) -> Self {
-        Error::GeneralError(e.to_string())
-    }
-}
-
-impl From<JsonPathError> for Error {
-    fn from(e: JsonPathError) -> Self {
-        Error::GeneralError(format!("JSON path error: {:?}", e).replace("\n", "\\n"))
+impl From<jsonpath_lib::JsonPathError> for Error {
```

Comment:
I was curious why you didn't use `#[from]` for this as well, and understood (by trying) that the reason is that `jsonpath_lib::JsonPathError` doesn't `impl std::error::Error`. Is that right?

If so, please add a comment that explains this for the next person who wonders.

Also: As long as the manual `From` is needed, I think it's cleaner to leave the `\n` escaping here instead of stuffing it into the `#[error] above.

Reply:
> jsonpath_lib::JsonPathError doesn't impl std::error::Error.
@yoav-steinberg let's open a PR on jsonpath_lib to change it

Reply:
Issued pr: https://github.com/freestrings/jsonpath/pull/52